huxley/core/tests/models/test_models.py

```diff
@@ -92,27 +92,44 @@ def test_update_country_preferences(self):
 
     def test_update_fees(self):
         '''Fees should be calculated when a School is created/updated.'''
-        b, i, a = 3, 5, 7
-        school = TestSchools.new_school(
-            beginner_delegates=b,
-            intermediate_delegates=i,
-            advanced_delegates=a,
-        )
 
+        school = TestSchools.new_school()
+        cm1 = TestCommittees.new_committee(name='CM1').id
+        cm2 = TestCommittees.new_committee(name='CM2').id
+        ct1 = TestCountries.new_country(name='CT1').id
+        ct2 = TestCountries.new_country(name='CT2').id
+        ct3 = TestCountries.new_country(name='CT3').id
+
+        #tests before assignments given out
         self.assertEquals(
             school.fees_owed,
-            school.REGISTRATION_FEE + school.DELEGATE_FEE * (b + i + a),
+            school.REGISTRATION_FEE,
         )
 
-        b2, i2, a2 = 5, 10, 15
-        school.beginner_delegates = b2
-        school.intermediate_delegates = i2
-        school.advanced_delegates = a2
+        Assignment.objects.bulk_create([
+            Assignment(committee_id=cm, country_id=ct, school_id=school.id, is_double=False)
+            for ct in [ct1, ct2]
+            for cm in [cm1, cm2]
+        ])
+
+        #test if fee is calculated using assingments 
         school.save()
```

Comment:
By saving the school after we add/update assignments this implies that every time we add/update assignments we must save the school in order to recalculate its fees. It might make more sense to recalculate the fees automatically any time an assignment is added or deleted. Some ways of doing this is signaling after an Assignment object is created or deleted to recalculate a school's fees. Or in the update_assignment functions modify it to call the update_fees method in the School class directly.